final release/format_data.py

```diff
@@ -0,0 +1,208 @@
+
+    #plt.scatter(first_plot.ra.deg, first_plot.dec.deg, label="first")
+    #plt.scatter(coord_asu.ra.deg, coord_asu.dec.deg, label="asu", alpha=0.5)
+    #plt.scatter(coord_fri.ra.deg, coord_fri.dec.deg, label="fri")
+    #plt.scatter(coord_frii.ra.deg, coord_frii.dec.deg, label="frii", alpha=0.5)
+    #plt.legend()
+    #plt.show()
+
+    # Finding nearest FRI's and FRIIs to each object in ASU
+    idx_i, sep_i, _i = coord_fri.match_to_catalog_sky(coord_asu)
+    idx_ii, sep_ii, _ii = coord_frii.match_to_catalog_sky(coord_asu)
+
+    # For sources within our set threshold, we assume that they are the same object.
+    fri_in_asu = (sep_i <= 5*u.arcsec)
+    frii_in_asu = (sep_ii <= 5*u.arcsec)
+
+    # Extacting FR images that are in asu
+    asu_fri = fri_images[fri_in_asu]
+    asu_frii = frii_images[frii_in_asu]
+
+    leftover_fri = fri_images[np.invert(fri_in_asu)]
+    leftover_frii = frii_images[np.invert(frii_in_asu)]
+
+    #Generating Labels
+    asu_fri_lab = np.zeros(asu_fri.shape[0], dtype=bool)
+    asu_frii_lab = np.ones(asu_frii.shape[0], dtype=bool)
+
+    leftover_fri_lab = np.zeros(leftover_fri.shape[0], dtype=bool)
+    leftover_frii_lab = np.zeros(leftover_frii.shape[0], dtype=bool)
+
+    # Combining FRI's and FRII's and shuffling
+    asu_fr = np.vstack((asu_fri, asu_frii))
+    asu_fr_lab = np.concatenate((asu_fri_lab, asu_frii_lab), axis=0) 
+
+    leftover_fr = np.vstack((leftover_fri, leftover_frii))
+    leftover_fr_lab = np.concatenate((leftover_fri_lab, leftover_frii_lab), axis=0)             
+
+    #Shuffling
+    asu_fr_, asu_fr_lab_ = shuffle(asu_fr, asu_fr_lab, random_state = seed)  
+    leftover_fr, leftover_fr_lab = shuffle(leftover_fr, leftover_fr_lab, random_state = seed) 
+
+    # Setting training and testing sets
+    if flip_train_test == False:
+        train_x = asu_fr_
+        train_y = asu_fr_lab_
+
+        test_x = leftover_fr
+        test_y = leftover_fr_lab
+
+
+    elif flip_train_test == True:
+        train_x = leftover_fr
+        train_y = leftover_fr_lab
+
+        test_x = asu_fr_
+        test_y = asu_fr_lab_
+
+    
+    # Defining rescaling function
+    def rescale_imgs(images,dim=dim):
+        size = images.shape[0]
+        imgs = np.zeros((size, dim, dim))
+
+        for i in range(size):
+            imgs[i, :, :] = resize(images[i, :, :], (dim, dim))
+
+        return np.expand_dims(imgs, axis=3)
+
+    
+    
+
+    #Rescaling Images
+    sys.stdout = open(os.devnull, "w") # don't show printed output
+    
+    train_x = rescale_imgs(train_x, dim)
+    test_x = rescale_imgs(test_x, dim)
+
+    sys.stdout = sys.__stdout__ # show printed output
+
+    # Convert class vectors to binary class matrices (for Keras Dense)
+    if conf_matrix == True:
+        train_y = np_utils.to_categorical(train_y, 2)
+        test_y = np_utils.to_categorical(test_y, 2)
+
+        
+    return train_x, train_y, test_x, test_y    
+
+
+import numpy as np
```

Comment:
Do these imports at the start of the file, not now.

Reply:
ok